wrappers/python/simtk/openmm/app/desmonddmsfile.py

```diff
@@ -212,10 +369,107 @@ def createSystem(self, nonbondedMethod=ff.NoCutoff, nonbondedCutoff=1.0*nanomete
         nb.setNonbondedMethod(methodMap[nonbondedMethod])
         nb.setCutoffDistance(nonbondedCutoff)
         nb.setEwaldErrorTolerance(ewaldErrorTolerance)
+        if cnb is not None:
+            nb.setUseDispersionCorrection(False)
+            if nonbondedMethod in (ff.CutoffPeriodic, ff.Ewald, ff.PME, ff.LJPME):
+                cnb.setNonbondedMethod(methodMap[ff.CutoffPeriodic])
+                cnb.setCutoffDistance(nonbondedCutoff)
+            elif nonbondedMethod == ff.CutoffNonPeriodic:
+                cnb.setNonbondedMethod(methodMap[ff.CutoffNonPeriodic])
+                cnb.setCutoffDistance(nonbondedCutoff)
+            else:
+                cnb.setNonbondedMethod(methodMap[ff.NoCutoff])
+            cnb.setUseSwitchingFunction(False)
+            cnb.setUseLongRangeCorrection(False)
+
+        #add implicit solvent model.
+        if implicitSolvent is not None:
+
+            if self._verbose:
+                print('Adding implicit solvent ...')
+
+            #with implicit solvent turn off native reaction field
+            #However note that this does not affect the shifted Coulomb potential of the Nonbonded force
+            #(it affects the only the energy, not the forces and equation of motion)
+            nb.setReactionFieldDielectric(1.0)
+
+            if implicitSolvent is HCT:
+                gb_parms = self._get_gb_params()
+
+                if gb_parms:
+                    if self._verbose:
+                        print('Adding HCT GB force ...')
+                    gb = GBSAHCTForce(SA='ACE')
+                    for i in range(len(gb_parms)):
+                        gb.addParticle(list(gb_parms[i]))
+                    gb.finalize()
+                    sys.addForce(gb)
+
+            if implicitSolvent is 'AGBNP3':
+                #load AGBNP3 plugin if available
+                try:
+                    from AGBNP3plugin import AGBNP3Force
+                    AGBNP3enabled = True
+                except ImportError:
+                    AGBNP3enabled = False
```

Comment:
If the user specifically requested a solvation model that isn't available, that should produce an exception.  Otherwise it's very easy for people to think they're simulating something totally different from what they really are.